machine/v1beta1/types_vsphereprovider.go

```diff
@@ -70,6 +70,13 @@ type VSphereMachineProviderSpec struct {
 	// When using LinkedClone, if no snapshots exist for the source template, falls back to FullClone.
 	// +optional
 	CloneMode CloneMode `json:"cloneMode,omitempty"`
+	// Disks is a list of non OS disks to be attached to the VM.
+	// +openshift:enable:FeatureGate=VSphereMultiDisk
+	// +openshift:validation:FeatureGateAwareMaxItems:featureGate="",maxItems=0
+	// +openshift:validation:FeatureGateAwareMaxItems:featureGate=VSphereMultiDisk,maxItems=15
+	// +kubebuilder:validation:XValidation:rule="self == oldSelf | size(self) == size(oldSelf)",message="additional disks cannot be added or removed once set"
```

Comment:
What does this actually mean? This is part of the vSphere providerSpec on the Machine right? So, we don't expect it to be changed once the machine is created anyway, that's true of most fields

When this is present on a MachineSet though, then it should be mutable right?

Reply:
Yes that is correct.  I expect machines to be not mutable, but for machineset, I'll follow the patterns of other fields in the machine def section.